backend/src/controllers/master.controller.ts

```diff
@@ -1,46 +1,50 @@
+import { CityMaster } from './../models/cityMaster.model';
 import { AddMasterSchema, DeleteMasterSchema, GetMasterSchema, UpdateMasterSchema, GetFreeMastersSchema } from './../validationSchemas/master.schema';
-import { Order } from './../models/order.model';
-import { sequelize } from './../sequelize';
+import { Order, WATCH_SIZES } from './../models/order.model';
 import { Master } from './../models/master.model';
 import { Request, Response } from 'express';
 import { Op } from 'sequelize';
-import { City } from '../models/city.model';
+import { sequelize } from '../sequelize';
 
 export default class MasterController {
     async addMaster(req: Request, res: Response): Promise<Response> {
+        const addMasterTransaction = await sequelize.transaction();
         try {
             const { name, cities } = AddMasterSchema.parse(req.body);
 
-            const existCities = await City.findAll({
-                where: {
-                    id: cities
-                }
+            const master = await Master.create({ name }, {
+                transaction: addMasterTransaction
             });
-            if(existCities.length !== cities.length) return res.status(404).json('Some of the cities does not exist')
-
-            const master = await Master.create({ name, cities });
+            await cities.reduce(async (promise, city) => {
```

Comment:
https://sebhastian.com/sequelize-belongstomany/#:~:text=The%20Sequelize%20belongsToMany()%20method%20is%20used%20to%20create%20a,primary%20keys%20of%20both%20models.

you can use method for making association